lms/djangoapps/django_comment_client/permissions.py

```diff
@@ -65,11 +65,27 @@ def check_question_author(user, content):
         if not content:
             return False
         try:
+            request_cache_dict = RequestCache.get_request_cache().data
             if content["type"] == "thread":
-                return content["thread_type"] == "question" and content["user_id"] == str(user.id)
+                cache_key = "django_comment_client.permissions._check_condition.check_question_author.{}.{}".format(
+                    user.id, content['id']
+                )
+                if cache_key in request_cache_dict:
+                    return request_cache_dict[cache_key]
+                else:
+                    result = content["thread_type"] == "question" and content["user_id"] == str(user.id)
+                    request_cache_dict[cache_key] = result
+                    return result
             else:
```

Comment:
For me edification, what else could `content['type']` be here, besides "thread"?

Reply:
I think `comment` (or `response`, maybe?) is the only other type, IIRC. It differentiates between the OP and responses to the OP

Reply:
Ah, right right right.